manager/state/raft/membership/cluster.go

```diff
@@ -180,7 +181,7 @@ func (c *Cluster) ReplaceMemberConnection(id uint64, oldConn *Member, newConn *M
 		return ErrIDNotFound
 	}
 
-	if oldConn.Conn != oldMember.Conn {
+	if !force && oldConn.Conn != oldMember.Conn {
```

Comment:
This part is not related to the address. It's about replacing the GRPC connection with a new connection. Imagine the address never changed, but an RPC failed. Instead of depending on GRPC's slow backoff, we close that connection and initiate a new one. This function replaces the old connection with the new one in the `Cluster` object.

Looking at the code before these changes might make it clearer.